lib/bounce_email.rb

```diff
@@ -212,15 +217,29 @@ def check_if_bounce(mail)
       false
     end
 
-    def get_original_mail(mail) #worked alright for me, for sure this as to be extended
-      parts = mail.body.to_s.split(/--- Below this line is a copy of the message.(\r\n\r\n|\n\n)/)
-      if parts.size > 1
-        ::Mail.new(parts.last)
-      elsif mail.parts
-        ::Mail.new(mail.parts[2].body)
+    def get_original_mail(mail) #worked alright for me, for sure this has to be extended
+      if mail.multipart?
+        ::Mail.new(mail.parts.last)
+      elsif i = index_of_original_message_delimiter(mail)
+        ::Mail.new(extract_original_message_after(mail, i))
+      else
+        nil
       end
     rescue => e
+      puts e
       nil
     end
+
+    def index_of_original_message_delimiter(mail)
+      INLINE_MESSAGE_DELIMITERS.find_index { |delimiter| self.body.to_s.include? delimiter }
```

Comment:
This check for delimiter inclusion seems too loose, esp. with a generic delimiter like "Original message". Maybe we should use the same regex you've used in the `split` call in `extract_original_message_after`.